src/main/java/net/kaczmarzyk/spring/data/jpa/web/SimpleSpecificationResolver.java

```diff
@@ -20,31 +20,39 @@
 import net.kaczmarzyk.spring.data.jpa.utils.QueryContext;
 import net.kaczmarzyk.spring.data.jpa.web.annotation.Spec;
 import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.EmbeddedValueResolver;
+import org.springframework.context.support.AbstractApplicationContext;
 import org.springframework.core.convert.ConversionService;
 import org.springframework.data.jpa.domain.Specification;
+import org.springframework.expression.ParseException;
 
 import java.lang.annotation.Annotation;
 import java.lang.reflect.InvocationTargetException;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collection;
 import java.util.regex.Pattern;
 
+import static java.util.Arrays.asList;
+
 
 /**
  * @author Tomasz Kaczmarzyk
  * @author Jakub Radlica
  */
 class SimpleSpecificationResolver implements SpecificationResolver<Spec> {
-
+	
 	private final ConversionService conversionService;
+	private final EmbeddedValueResolver embeddedValueResolver;
 	
-	public SimpleSpecificationResolver(ConversionService conversionService) {
+	public SimpleSpecificationResolver(ConversionService conversionService, AbstractApplicationContext applicationContext) {
```

Comment:
SHoulnd't we keep a separate contrstructor with just 1 param of type ConversionService?